scripts/build_llvm_project.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import platform
+import settings
+import shutil
+import subprocess
+import sys
+
+from pathlib import Path, PurePath
+
+def default_toolchain():
+  system = platform.uname()[0]
+  machine = platform.uname()[4]
+  toolchain = Path(settings.THIS_DIR,
+                           'cmake',
+                           f'toolchain_{system.lower()}_{machine.lower()}.cmake')
+  return toolchain if toolchain.is_file else None
+
+def get_arguments():
+  parser = argparse.ArgumentParser(
+    description="Help configure and build classic-flang-llvm-project")
+
+  buildopt = parser.add_argument_group('general build options')
+  buildopt.add_argument('-t', '--target', metavar='ARCH', choices=['X86', 'AArch64', 'PowerPC'], default='X86',
+                        help='Control which targets are enabled (%(choices)s) (default: %(default)s)')
+  buildopt.add_argument('-p', '--install-prefix', metavar='PATH', nargs='?', default=Path(settings.LLVM_DIR,'..','install'), const=False,
+                        help='install directory (default: %(default)s) ')
+  buildopt.add_argument('-j', '--jobs', metavar='N', type=int, default=os.cpu_count(),
+                        help='number of parallel build jobs (default: %(default)s)')
+  buildopt.add_argument('--toolchain', metavar='FILE', default=default_toolchain().as_posix(),
+                        help='specify toolchain file (default: %(default)s)')
+  buildopt.add_argument('--builddir', metavar='DIR', default='build',
+                        help=f'specify build directory (default: {settings.LLVM_DIR}/%(default)s)')
+  buildopt.add_argument('--clean', action='store_true', default=False,
+                        help='clean build')
+  buildopt.add_argument('-d', '--build-type', metavar='TYPE', default='Release',
+                        help='set build type (default: %(default)s)')
+  buildopt.add_argument('-x', '--cmake-param', metavar='OPT', action='append', default=[],
+                        help='add custom argument to CMake')
+  buildopt.add_argument('-e', '--llvm-enable-projects', metavar='OPT', action='append', default=['clang'] if sys.platform == 'win32' else ['clang;openmp'],
+                        help='enable llvm projects to build (in quotation marks separated by semicolons e.g.: "clang;openmp")')
+  buildopt.add_argument('-c', '--use-ccache', action="store_true", default=False,
+                        help='Using ccache during the build (default: %(default)s)')
+  buildopt.add_argument('-a', '--c-compiler', metavar='OPT', default=None,
```

Comment:
`-a` was never a good mnemonic for this option. I also don't think this will be used very often since we can use `--toolchain`. I would suggest renaming this to `--cc`, without a mnemonic.